sda-download/internal/config/config.go

```diff
@@ -380,9 +376,17 @@ func (c *Map) appConfig() error {
 	}
 
 	var err error
-	c.App.Crypt4GHPrivateKey, c.App.Crypt4GHPublicKeyB64, err = GenerateC4GHKey()
-	if err != nil {
-		return err
+	if viper.GetString("app.c4gh.privateKeyPath") != "" {
+
+		if !viper.IsSet("app.c4gh.passphrase") {
+			return errors.New("app.c4gh.passphrase is not set")
+		}
```

Comment:
```suggestion
	if viper.GetString("app.c4gh.privateKeyPath") != "" {

		if !viper.IsSet("c4gh.passphrase") {
			return errors.New("c4gh.passphrase is not set")
		}
```

c4gh config is already declared and we only need one entry in the config

Reply:
no it is not, there is no other entry for c4gh keys in `sda-download/internal/config.go`

Reply:
All the issues stems from the fact that the config.yaml is not dedicated to sda-download.
Still, it should not be under the `app` section since that holds the HTTP relates configs for port, tls and so.

Reply:
I think the reason why @aaperis put it under `app` instead of directly under `c4gh` is try to distinguish this private key from the private key used for other services such as `ingest`. There might be some confusing when we try to merge `sda-download` to SDA later on. I'm OK to put at either places as long as they works. But I prefer that they are distinguishable.

Reply:
As stated elsewhere, do not mix in how the other sda services does things.
There is zero code overlap at the moment and if one would make sure that the issues are kept to a minimum when this gets merged the make sure to follow the given semantics on how things are structured.

Reply:
I would prefer to keep the variable names separated. I don't find it practical to use variable naming that other sda applications use only to change them in the future because they overlap.

Reply:
Keeping the names unique is a no-brainer since it will make things easier down the road, but it should stil be under the `c4gh` heading to conform with the overall semantics.

So something like this maybe:
```
c4gh.transientKeyPath
c4gh.transientPassphrase
```